File: biometric_attendance/biometric_attendance/report/attendance_rules/attendance_rules.js
// Copyright (c) 2016, Akshay Mehta and contributors
// For license information, please see license.txt
/* eslint-disable */

frappe.query_reports["Attendance Rules"] = {
	"filters": [
		{
		  "fieldname": "from_date",
		  "fieldtype": "Date",
		  "label": "From Date",
		  "default": frappe.datetime.add_months(frappe.datetime.month_start())
		},
		{
		  "fieldname": "to_date",
		  "fieldtype": "Date",
		  "label": "To Date",
		  "default": frappe.datetime.add_months(frappe.datetime.month_end())
		},
		{
		  "fieldname": "user",
		  "fieldtype": "Link",
		  "label": "User",
		  "options": "Biometric Users",
		  "reqd": 1,
		  "on_change": function() {
			var user = frappe.query_report.get_filter_value("user");
			console.log("user ----------"+user)
			if (user) {
				frappe.db.get_value("Biometric Users", user, "user_name", function(value) {
					console.log("---------------------"+value["user_name"])
					frappe.query_report.set_filter_value("user_name",value["user_name"]);
				});
			}
		  }
		},
		{
		  "fieldname": "user_name",
		  "fieldtype": "Data",
		  "label": "User Name",
		  "read_only": 1
		}
	]
}

File: biometric_attendance/biometric_attendance/report/attendance_rules/attendance_rules.py
# Copyright (c) 2013, Akshay Mehta and contributors
# For license information, please see license.txt

from __future__ import unicode_literals
import frappe

def execute(filters=None):
	columns, data = [], []
	user_id = filters.get("user")
	from_date = filters.get("from_date")
	to_date = filters.get("to_date")
	from_date = str(from_date)+" 00:00:00"
	to_date = str(to_date)+" 23:59:59"
	attendance_record = frappe.db.sql(""" select * from `tabBiometric Attendance` where user_id = %s and timestamp >= %s and timestamp <= %s and docstatus=1""",(user_id,from_date,to_date), as_dict=1)
	#for att in attendance_record:
	
	return columns, data
